tests/test_sanity.py

```diff
@@ -116,12 +116,75 @@ def test_nodejs(
     gprofiler_docker_image: Image,
 ) -> None:
     with SystemProfiler(
-        1000, 6, Event(), str(tmp_path), False, perf_mode="fp", perf_inject=True, perf_dwarf_stack_size=0
+        1000,
+        6,
+        Event(),
+        str(tmp_path),
+        False,
+        perf_mode="fp",
+        perf_inject=True,
+        perf_dwarf_stack_size=0,
+        perf_node_attach=False,
     ) as profiler:
         process_collapsed = snapshot_pid_collapsed(profiler, application_pid)
         assert_collapsed(process_collapsed)
 
 
+@pytest.mark.parametrize("profiler_type", ["attach-maps"])
+@pytest.mark.parametrize("runtime", ["nodejs"])
+@pytest.mark.parametrize("application_image_tag", ["without-flags"])
+@pytest.mark.parametrize("command_line", [["node", f"{CONTAINERS_DIRECTORY}/nodejs/fibonacci.js"]])
+def test_nodejs_attach_maps(
+    tmp_path: Path,
+    application_pid: int,
+    assert_collapsed: AssertInCollapsed,
+    profiler_type: str,
+    command_line: List[str],
+    runtime_specific_args: List[str],
+) -> None:
+    with SystemProfiler(
+        1000,
+        6,
+        Event(),
+        str(tmp_path),
+        False,
+        perf_mode="fp",
+        perf_inject=False,
+        perf_dwarf_stack_size=0,
+        perf_node_attach=True,
+    ) as profiler:
+        process_collapsed = snapshot_pid_collapsed(profiler, application_pid)
+        assert_collapsed(process_collapsed)
+        # check for node built-in functions
+        assert_function_in_collapsed("node::Start", process_collapsed)
+        # check for v8 built-in functions
+        assert_function_in_collapsed("v8::Function::Call", process_collapsed)
+
+
+@pytest.mark.parametrize("profiler_type", ["attach-maps"])
+@pytest.mark.parametrize("runtime", ["nodejs"])
+@pytest.mark.parametrize("application_image_tag", ["without-flags"])
+@pytest.mark.parametrize("command_line", [["node", f"{CONTAINERS_DIRECTORY}/nodejs/fibonacci.js"]])
+def test_nodejs_attach_maps_from_container(
+    docker_client: DockerClient,
+    application_pid: int,
+    runtime_specific_args: List[str],
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    assert_collapsed: AssertInCollapsed,
+    assert_app_id: Callable,
```

Comment:
We can keep it although it doesn't do much as we don't have appid support for nodejs yet (https://github.com/Granulate/gprofiler/issues/504)